queue/src/main/java/org/killbill/notificationq/DefaultNotificationQueue.java

```diff
@@ -327,6 +337,44 @@ public Void withSqlDao(final NotificationSqlDao transactional) throws Exception
         InTransaction.execute(dbi, connection, handler, NotificationSqlDao.class);
     }
 
+    @Override
+    public void removeFutureNotificationsForSearchKeys(final Long searchKey1, final Long searchKey2) {
+        dao.getSqlDao().inTransaction(new Transaction<Void, QueueSqlDao<NotificationEventModelDao>>() {
+            @Override
+            public Void inTransaction(final QueueSqlDao<NotificationEventModelDao> transactional, final TransactionStatus status) throws Exception {
+                // Move entries by batch into the history table
+                final int batchSize = 25;
+                final Collection<NotificationEventModelDao> currentBatch = new ArrayList<NotificationEventModelDao>(batchSize);
+
+                // Note that we don't claim them here, so it could be possible that some of these entries end up being processed nonetheless
```

Comment:
But then, wouldn't we get an exception when the thread processing the entry is trying to move the row into `history`. Is this api only used for tests ?

Reply:
> But then, wouldn't we get an exception when the thread processing the entry is trying to move the row into `history`.

I don't think so, as `moveEntriesToHistory` is really:

1. Add the entries from memory (already selected) into the history table
2. Delete by `record_id` in the main table

So in the worst case scenario, we could have duplicates in the history table (2 rows with 2 different states, `PROCESSED` and `REMOVED`), but I don't think we care?

> Is this api only used for tests ?

No, this would be called by the `closeAccount` endpoint.